### PR TITLE
status hashtags, android: fix #1247

### LINKED ISSUE #1247: Tag is shown as usual text if it contains non-English chars (e.g. #слово) [Android only]
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: If tag contains non-English chars then it's shown as plain text instead of blue text. It happens for 
- Recent list in Discover screen
- tag in Popular tags section on Discover screen
- text inside Status field in user profile

screenshot_20170527-124223

screenshot_20170527-132632

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
Word with # should be highlighted with blue colour (as it happens for tags with English text)
#### Actual behavior
[comment]: # (Describe what actually happened.)
Tags with Russian text or other non-English letters are shown as plain text. If status contains English tags and non-English ones then all of them are shown a plain text.
### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

- Open Status
- Open profile and edit it
- in status type #слово and save it. Check if it's shown with blue colour
- open Discover tab and check in Recents and Popular tags how tag is shown

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: 0.9.7
* Operating System: Android only. Android 6.0.1, Galaxy S6

### PR DESCRIPTION
**Pull request by tiensonqin**
fix #1247

### Summary:

`(.startsWith "#你好" "#")` not works, but both
`(.startsWith "#你好" "#你")` and `(.startsWith "#abc" "#")` works.

### CODE REVIEW

File: src/status_im/android/platform.cljs [hunk 1]
```diff
@@ -37,8 +37,7 @@
                                       :background-color "#7099e619"
                                       :border-radius    5
                                       :padding          4}
-                           :item     {:status-text {:color       styles/color-black
```
Comment (rasom) [inline src/status_im/android/platform.cljs]: It's not clear how this change is related to PR's objective, could you elaborate please, @tiensonqin?

File: src/status_im/components/drawer/styles.cljs [hunk 2]
```diff
@@ -52,10 +52,12 @@
    :color               common/color-black})
 
 (defnstyle status-view [placeholder?]
-  (merge status-input-view
-         {:color      (if placeholder? common/color-gray common/color-black)
-          :min-height 0
-          :ios        {:padding-top 10}}))
+  (-> (dissoc status-input-view :color)
+      (cond-> placeholder?
+        (assoc :color common/color-gray))
```
Comment (rasom) [inline src/status_im/components/drawer/styles.cljs]: The same here, we are excluding `common/color-black`, but why?

File: src/status_im/android/platform.cljs (hunk 1 above)
Comment (tiensonqin) [inline src/status_im/android/platform.cljs]: @rasom Hi Roman, thanks for your quick review!
The reason is that `line-height` not works very well with `color` on Android.
I just set up a minimal example using Expo: 
https://github.com/tiensonqin/text-style-example/blob/master/src/text_style_example/core.cljs#L10

Comment (rasom) [inline src/status_im/android/platform.cljs]: sorry, but I still don't understand what is the problem there? Could you explain? `line-height` doesn't work when `color` is used, or vise versa? Or what happens?

**Comment by rasom** [comment]
Please, rebase PR on `develop` branch and squash commits

**Comment by tiensonqin** [comment]
Yeah, in my test case color doesn't work well when line-height is used.

File: src/status_im/components/status_view/view.cljs [hunk 3]
```diff
@@ -4,26 +4,30 @@
             [clojure.string :as str]
             [status-im.components.react :refer [view text]]
             [status-im.utils.platform :refer [platform-specific]]
-            [status-im.components.styles :refer [color-blue]]))
+            [status-im.components.styles :refer [color-blue]]
+            [status-im.utils.utils :refer [hash-tag?]]
+            [status-im.components.styles :as styles]))
```
Comment (flexsurfer) [inline src/status_im/components/status_view/view.cljs]: double styles

**Comment by rasom** [comment]
Works as expected, or i'm missing something?

```clojure
[view {:style {:margin-top 30}}
     [text {:style {:color "blue"
                    :line-height 21}}
      "blue "
      [text {:style {:color "red"}}
       "red "]
      [text {:style {:color "green"}}
       "green"]]]
```

screen shot 2017-07-03 at 15 30 22

File: src/status_im/components/status_view/view.cljs [hunk 4]
```diff
@@ -4,26 +4,30 @@
...
+                    :or {message-id "msg"
+                         non-tag-color styles/color-black}}]
   [text {:style           style
          :on-press        on-press
          :number-of-lines number-of-lines
          :font            :default}
    (for [[i status] (map-indexed vector (str/split status #" "))]
-     (if (.startsWith status "#")
-       ^{:key (str "item-" message-id "-" i)}
+     ^{:key (str "item-" message-id "-" i)}
```
Comment (flexsurfer) [inline src/status_im/components/status_view/view.cljs]: this form will not work,  `with-meta` should be used

File: src/status_im/components/status_view/view.cljs (hunk 3 above)
Comment (tiensonqin) [inline src/status_im/components/status_view/view.cljs]: Thanks for pointing this out, I should be more careful.

File: src/status_im/components/status_view/view.cljs (hunk 4 above)
Comment (tiensonqin) [inline src/status_im/components/status_view/view.cljs]: I really don't know this, the apps runs fine, I'll examine this later.

**Comment by tiensonqin** [comment]
@rasom Not works on Android in my case, https://github.com/tiensonqin/status-react.
screenshot_20170703-203640

**Comment by rasom** [comment]
@tiensonqin let's not touch these styles. If problem will be there, then we will try to fix it, ok?

File: src/status_im/components/status_view/view.cljs (hunk 4 above)
Comment (tiensonqin) [inline src/status_im/components/status_view/view.cljs]: I thought clojure compiler will replace `^{:key k} expr` with `(with-meta {:key k} expr)`.

**Comment by tiensonqin** [comment]
@rasom 
The problem is not related to styles stuff.
In this case, `red` should be red and `green` should be green since what we expect is the children styles will override the parent styles, but they don't work here(only on Android).
https://github.com/tiensonqin/text-style-example/blob/master/src/text_style_example/core.cljs#L8-L15

**Comment by rasom** [comment]
@tiensonqin i tested this inside status-react app, and don't see any problem https://github.com/status-im/status-react/pull/1398#issuecomment-312633161

**Comment by tiensonqin** [comment]
@rasom That's weird, it doesn't work on both the Expo example and status-react app for me.

**Comment by tiensonqin** [comment]
I only tested this on two real devices:
1. Android 7.0, Nexus 6
2. Android 7.1.1, OnePlus 3T

**Comment by rasom** [comment]
Could you please test this again with code in `status-react` (with code related to this PR) and show screenshots with the problem? Thanks

**Comment by tiensonqin** [comment]
@rasom I just run using the latest `develop` branch, the problem still exists.
screenshot_20170703-221544
screenshot_20170703-221525

**Comment by rasom** [comment]
Thanks for explanation @tiensonqin. I still can't reproduce this on my device, but as it works as expected changes will be merged.

**Pull request merged**